docs/experiments-msmarco-irst.md

```diff
@@ -0,0 +1,226 @@
+  --query_path irst_test/queries.irst_topics.dev.small.json \
+  --index ~/.cache/pyserini/indexes/index-msmarco-passage-ltr-20210519-e25e33f.a5de642c268ac1ed5892c069bdc29ae3/ \
+  --output irst_test/regression_test_sum.txt \
+  --alpha 0.1
+```
+
+IRST (Max)
+```bash
+python -m pyserini/scripts/reranker_ibm_colbert.py 
+  --base runs/run.topics.bm25tuned.txt
+  --tran_path irst_test/ibm_model_1_bert_tok_20211117/ \
+  --query_path irst_test/queries.irst_topics.dev.small.json \
+  --index ~/.cache/pyserini/indexes/index-msmarco-passage-ltr-20210519-e25e33f.a5de642c268ac1ed5892c069bdc29ae3/ \
+  --output irst_test/regression_test_max.txt \
+  --alpha 0.3 \
+  --max_sim
+```
+
+For different topics, the `--input`,`--topics` and `--qrel` are different, since Pyserini has all these topics available, we can pass in
+different values to run on different datasets.
+
+`--input`: <br />
+&nbsp;&nbsp;&nbsp;&nbsp;TREC DL 2019 Passage: `tools/topics-and-qrels/topics.dl19-passage.txt` <br />
+&nbsp;&nbsp;&nbsp;&nbsp;TREC DL 2020 Passage: `tools/topics-and-qrels/topics.dl20.txt` <br />
+&nbsp;&nbsp;&nbsp;&nbsp;MS MARCO Passage V1: `tools/topics-and-qrels/topics.msmarco-passage.dev-subset.txt` <br />
+
+`--topics`: <br />
+&nbsp;&nbsp;&nbsp;&nbsp;TREC DL 2019 Passage: `dl19-passage` <br />
+&nbsp;&nbsp;&nbsp;&nbsp;TREC DL 2020 Passage: `dl20` <br />
+&nbsp;&nbsp;&nbsp;&nbsp;MS MARCO Passage V1: `msmarco-passage-dev-subset` <br />
+
+
+
+After the run finishes, we can also evaluate the results using the official MS MARCO evaluation script:
+
+```bash
+tools/eval/trec_eval.9.0.4/trec_eval -c -m ndcg_cut -m map -m recip_rank qrel_file irst_test/regression_test_sum.txt
+```
+
+`--qrel_file`: <br />
+&nbsp;&nbsp;&nbsp;&nbsp;TREC DL 2019 Passage: `tools/topics-and-qrels/qrels.dl19-passage.txt` <br />
+&nbsp;&nbsp;&nbsp;&nbsp;TREC DL 2020 Passage: `tools/topics-and-qrels/qrels.dl20-passage.txt` <br />
+&nbsp;&nbsp;&nbsp;&nbsp;MS MARCO Passage V1: `tools/topics-and-qrels/qrels.msmarco-passage.dev-subset.txt` <br />
+
+
+Note that we evaluate MRR and NDCG at a cutoff of 10 hits to match the official evaluation metrics.
+
+
+
+## Document Reranking 
+
+
+### Data Preprocessing
+
+For MSMARCO DOC, each MS MARCO document is first segmented into passages, each passage is treated as a unit of indexing. More details about indexing could be found in this anserini documentation: https://github.com/castorini/anserini/blob/master/docs/regressions-msmarco-doc-segmented.md.
+We utilized the MaxP technique during the ranking, that is scoring documents based on one of its highest-scoring passage.
+
+
+
+The below scripts convert queries to json objects with text, text_unlemm, raw, and text_bert_tok fields
+
+```bash
+mkdir irst_test
+python scripts/ltr_msmarco/convert_queries.py --input path_to_topics --output irst_test/queries.irst_topics.dev.small.json
+```
+
+We can now perform retrieval in anserini to generate baseline:
+
+```
+target/appassembler/bin/SearchCollection \
```

Comment:
This is anserini, not pyserini?

Reply:
Yes, this is in anserini, as the document is segmented. I couldn't find a proper way to index segmented-document in pyserini. But I guess I could still use pyserini to retrieval?

Reply:
Something like this? https://github.com/castorini/pyserini/blob/master/docs/pypi-reproduction.md#ms-marco-document-ranking